crates/rolldown/src/ast_scanner/impl_visit.rs

```diff
@@ -554,8 +559,8 @@ impl<'me, 'ast: 'me> AstScanner<'me, 'ast> {
     ident_ref: &IdentifierReference,
   ) -> Option<()> {
     let parent = self.visit_path.last()?;
-    if let AstKind::CallExpression(_) = parent {
-      if ident_ref.name == "eval" {
+    if let AstKind::CallExpression(call_expr) = parent {
+      if ident_ref.name == "eval" && call_expr.callee.address() == Address::from_ref(ident_ref) {
```

Comment:
https://github.com/oxc-project/oxc/pull/15702 is going to remove `Address::from_ref`. `Address::from_ptr` does the same thing.

```suggestion
      if ident_ref.name == "eval" && call_expr.callee.address() == Address::from_ptr(ident_ref) {
```

Ultimately, we'll want to migrate from `Address::from_ptr` to `UnstableAddress::unstable_address` (introduced in https://github.com/oxc-project/oxc/pull/15700), but we can do that once `unstable_address` has landed in an Oxc release.

Reply:
fixed.